spec/rubocop/cop/lint/debugger_spec.rb

```diff
@@ -87,120 +122,176 @@
+      expect_offense(<<~RUBY)
+        Kernel.binding.pry
+        ^^^^^^^^^^^^^^^^^^ Remove debugger entry point `Kernel.binding.pry`.
+      RUBY
+    end
 
-  it 'does not report an offense for a non-pry binding' do
-    expect_no_offenses('binding.pirate')
-  end
+    it 'registers an offense for a Pry.rescue call' do
+      expect_offense(<<~RUBY)
+        def method
+          Pry.rescue { puts 1 }
+          ^^^^^^^^^^ Remove debugger entry point `Pry.rescue`.
+          ::Pry.rescue { puts 1 }
+          ^^^^^^^^^^^^ Remove debugger entry point `::Pry.rescue`.
+        end
+      RUBY
+    end
 
-  it 'reports an offense for a debugger with Kernel call' do
-    expect_offense(<<~RUBY)
-      Kernel.debugger
-      ^^^^^^^^^^^^^^^ Remove debugger entry point `Kernel.debugger`.
-    RUBY
+    it 'does not register an offense for a `pry` call without binding' do
+      expect_no_offenses('pry')
+    end
+
+    it 'does not register an offense for a `rescue` call without Pry' do
+      expect_no_offenses(<<~RUBY)
+        begin
+        rescue StandardError
+        end
+      RUBY
+    end
   end
 
-  it 'reports an offense for a debugger with ::Kernel call' do
-    expect_offense(<<~RUBY)
-      ::Kernel.debugger
-      ^^^^^^^^^^^^^^^^^ Remove debugger entry point `::Kernel.debugger`.
-    RUBY
+  context 'rails' do
+    it 'registers an offense for a debugger call' do
+      expect_offense(<<~RUBY)
+        debugger
+        ^^^^^^^^ Remove debugger entry point `debugger`.
+      RUBY
+    end
+
+    it 'registers an offense for a debugger with an argument call' do
+      expect_offense(<<~RUBY)
+        debugger foo
+        ^^^^^^^^^^^^ Remove debugger entry point `debugger foo`.
+      RUBY
+    end
+
+    it 'registers an offense for a debugger with Kernel call' do
+      expect_offense(<<~RUBY)
+        Kernel.debugger
+        ^^^^^^^^^^^^^^^ Remove debugger entry point `Kernel.debugger`.
+      RUBY
+    end
+
+    it 'registers an offense for a debugger with ::Kernel call' do
+      expect_offense(<<~RUBY)
+        ::Kernel.debugger
+        ^^^^^^^^^^^^^^^^^ Remove debugger entry point `::Kernel.debugger`.
+      RUBY
+    end
   end
 
-  it 'reports an offense for a binding.pry with Kernel call' do
-    expect_offense(<<~RUBY)
-      Kernel.binding.pry
-      ^^^^^^^^^^^^^^^^^^ Remove debugger entry point `Kernel.binding.pry`.
-    RUBY
+  context 'web console' do
+    it 'registers an offense for a `binding.console` call' do
+      expect_offense(<<~RUBY)
+        binding.console
+        ^^^^^^^^^^^^^^^ Remove debugger entry point `binding.console`.
+      RUBY
+    end
+
+    it 'does not register an offense for `console` without a receiver' do
+      expect_no_offenses('console')
+    end
   end
 
-  it 'reports an offense for save_and_open_page with Kernel' do
-    expect_offense(<<~RUBY)
-      Kernel.save_and_open_page
-      ^^^^^^^^^^^^^^^^^^^^^^^^^ Remove debugger entry point `Kernel.save_and_open_page`.
-    RUBY
```

Comment:
This test was [discussed in #8929 ](https://github.com/rubocop-hq/rubocop/pull/8929#pullrequestreview-524890240) as being there because it was not possible to exclude it in the way the cop was changed, but capybara does not actually add the method to `Kernel` so it's a false positive that I've now removed.